runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -1079,9 +1081,729 @@ J9::AheadOfTimeCompile::dumpRelocationData()
             traceMsg(self()->comp(), "\n ClassUnloadAssumption \n");
             break;
 
+         case TR_ValidateRootClass:
+            {
+            cursor++;
+            if (is64BitTarget)
+               cursor += 4;     // padding
+            cursor -= sizeof(TR_RelocationRecordBinaryTemplate);
+            TR_RelocationRecordValidateRootClassBinaryTemplate *binaryTemplate =
+                  reinterpret_cast<TR_RelocationRecordValidateRootClassBinaryTemplate *>(cursor);
+            if (isVerbose)
+               {
+               traceMsg(self()->comp(), "\n Validate Root Class: classID=%d ", (uint32_t)binaryTemplate->_classID);
+               }
+            cursor += sizeof(TR_RelocationRecordValidateRootClassBinaryTemplate);
+            self()->traceRelocationOffsets(cursor, offsetSize, endOfCurrentRecord, orderedPair);
+            }
+            break;
+
+         case TR_ValidateClassByName:
+            {
+            cursor++;
+            if (is64BitTarget)
+               cursor += 4;     // padding
+            cursor -= sizeof(TR_RelocationRecordBinaryTemplate);
+            TR_RelocationRecordValidateClassByNameBinaryTemplate *binaryTemplate =
+                  reinterpret_cast<TR_RelocationRecordValidateClassByNameBinaryTemplate *>(cursor);
+            if (isVerbose)
+               {
+               traceMsg(self()->comp(), "\n Validate Class By Name: classID=%d beholderID=%d primitiveType=%c romClassOffsetInSCC=%p ",
+                        (uint32_t)binaryTemplate->_classID,
+                        (uint32_t)binaryTemplate->_beholderID,
```

Comment:
I'm not wedded to the name 'beholder' by any means; it came about because when @jdmpapin and I were discussing the semantics of the SVM, we kept talking about the idea of acquiring a symbol "as seen by" another symbol (eg Class By Name), and so "beholder" was the word we ended up using. Any name that conveys the idea that the symbol being validated depends on, or is acquired via, this other symbol is fine by me.